app/api/groups/get_user_descendants.feature

```diff
@@ -158,220 +66,233 @@ Feature: List user descendants of the group (groupUserDescendantView)
     """
     [
       {
-        "id": "63",
-        "name": "janeb",
-        "user": {
-          "first_name": "Jane",
-          "grade": null,
-          "id": "17",
-          "last_name": "Baker",
-          "login": "janeb"
-        },
-        "parents": [
-          {
-            "id": "12",
-            "name": "Other Class"
-          },
-          {
-            "id": "16",
-            "name": "First Team"
-          }
-        ]
+        "id": "51",
+        "name": "johna",
+        "parents": [{"id": "11", "name": "Our Class"}],
+        "user": {"first_name": null, "grade": 1, "id": "11", "last_name": "Adams", "login": "johna"}
       }
     ]
     """
 
-  Scenario: Get users skipping the first five
-    Given I am the user with ID "1"
-    When I send a GET request to "/groups/1/user-descendants?from.name=johnb&from.id=53"
+  Scenario: Only actual memberships count
+    Given the database table 'users' has also the following rows:
+      | ID | sLogin | idGroupSelf | idGroupOwned | sFirstName  | sLastName | iGrade |
+      | 11 | johna  | 51          | 52           | John        | Adams     | 1      |
+      | 12 | johnb  | 53          | 54           | John        | Baker     | 2      |
+      | 13 | johnc  | 55          | 56           | John        | null      | 3      |
+      | 14 | johnd  | 57          | 58           | null        | Davis     | -1     |
+      | 15 | johne  | 59          | 60           | John        | Edwards   | null   |
+      | 16 | janea  | 61          | 62           | Jane        | Adams     | 3      |
+      | 17 | janeb  | 63          | 64           | Jane        | Baker     | null   |
+    And the database table 'groups' has also the following rows:
+      | ID | sType     | sName          | iGrade |
+      | 51 | UserSelf  | johna          | -2     |
+      | 52 | UserAdmin | johna-admin    | -2     |
+      | 53 | UserSelf  | johnb          | -2     |
+      | 54 | UserAdmin | johnb-admin    | -2     |
+      | 55 | UserSelf  | johnc          | -2     |
+      | 56 | UserAdmin | johnc-admin    | -2     |
+      | 57 | UserSelf  | johnd          | -2     |
+      | 58 | UserAdmin | johnd-admin    | -2     |
+      | 59 | UserSelf  | johne          | -2     |
+      | 60 | UserAdmin | johne-admin    | -2     |
+      | 61 | UserSelf  | janea          | -2     |
+      | 62 | UserAdmin | janea-admin    | -2     |
+    And the database table 'groups_groups' has also the following rows:
+      | idGroupParent | idGroupChild | sType              |
+      | 11            | 51           | invitationSent     |
+      | 11            | 53           | requestSent        |
+      | 11            | 55           | invitationRefused  |
+      | 11            | 57           | requestRefused     |
+      | 11            | 59           | removed            |
+      | 11            | 61           | left               |
+    And the database table 'groups_ancestors' has also the following rows:
```

Comment:
Soon we should do a "the database table 'groups_ancestors' match 'groups_groups'", it would save many lines ;-)
It can be done later...

Reply:
> "the database table 'groups_ancestors' match 'groups_groups'"
wow! a great idea!